avalon/tools/contextmanager/app.py

```diff
@@ -211,19 +213,21 @@ def select_asset(self, assetname):
         self._assets.select_assets([assetname], expand=True)
 
 
-def show(parent=None):
+def create_window(parent):
+    window = App(parent)
+    window.show()
+    window.setStyleSheet(style.load_stylesheet())
+    module.window = window
 
-    from avalon import style
-    from ...tools import lib
-    try:
-        module.window.close()
-        del module.window
-    except (RuntimeError, AttributeError):
-        pass
 
+def show(parent=None):
     with lib.application():
-        window = App(parent)
-        window.show()
-        window.setStyleSheet(style.load_stylesheet())
+        if lib.existing_app:
+            try:
+                module.window.raise_()
+            except (RuntimeError, AttributeError):
+                create_window(parent)
+        else:
+            create_window(parent)
```

Comment:
Is this case tested? What do we expect to happen? If there isn't a `QApplication`, then we probably can't create a window.

Reply:
I've tested this in Nuke/Maya and standalone (Photoshop).

If there isn't a `QApplication` the `with lib.application()` ensure there is a `QApplication`.

Reply:
But isn't `lib.application()` happening first? :O How can `existing_app` be False, if `application()` ensures it is true?